include/deal.II/lac/petsc_ts.h

```diff
@@ -0,0 +1,706 @@
+      , minimum_step_size(minimum_step_size)
+      , maximum_step_size(maximum_step_size)
+      , absolute_tolerance(absolute_tolerance)
+      , relative_tolerance(relative_tolerance)
+    {}
+
+    /**
+     * Add all TimeStepperData() parameters to the given ParameterHandler
+     * object. When the parameters are parsed from a file, the internal
+     * parameters are automatically updated.
+     *
+     * These are one-to-one with the options you can pass at construction
+     * time.
+     *
+     * The options you pass at construction time are set as default values in
+     * the ParameterHandler object `prm`. You can later modify them by parsing
+     * a parameter file using `prm`. The values of the parameter will be
+     * updated whenever the content of `prm` is updated.
+     *
+     * Make sure that this class lives longer than `prm`. Undefined behavior
+     * will occur if you destroy this class, and then parse a parameter file
+     * using `prm`.
+     */
+    void
+    add_parameters(ParameterHandler &prm)
+    {
+      prm.enter_subsection("Running parameters");
+      prm.add_parameter("Solver type", tstype);
+      prm.add_parameter("Initial time", initial_time);
+      prm.add_parameter("Final time", final_time);
+      prm.add_parameter("Initial step size", initial_step_size);
+      prm.add_parameter("Maximum number of steps", max_steps);
+      prm.add_parameter("Match final time", match_step);
+      prm.leave_subsection();
+
+      prm.enter_subsection("Error control");
+      prm.add_parameter("Minimum step size", minimum_step_size);
+      prm.add_parameter("Maximum step size", maximum_step_size);
+      prm.add_parameter("Absolute error tolerance", absolute_tolerance);
+      prm.add_parameter("Relative error tolerance", relative_tolerance);
+      prm.leave_subsection();
+    }
+
+    /**
+     * PETSc solver type
+     */
+    std::string tstype;
+
+    /**
+     * Initial time for the DAE.
+     */
+    double initial_time;
+
+    /**
+     * Final time.
+     */
+    double final_time;
+
+    /**
+     * Initial step size.
+     */
+    double initial_step_size;
+
+    /**
+     * Maximum number of steps to be taken.
+     * Negative values are ignored.
+     */
+    int max_steps;
+
+    /**
+     * Match final time requested?
+     */
+    bool match_step;
+
+    /**
+     * Minimum allowed step size.
+     * Non-positive values indicate to use PETSc's default.
+     */
+    double minimum_step_size;
+
+    /**
+     * Maximum allowed step size.
+     * Non-positive values indicate to use PETSc's default.
+     */
+    double maximum_step_size;
+
+    /**
+     * Absolute error tolerance for adaptive time stepping.
+     * Negative values indicate to use PETSc's default.
+     */
+    double absolute_tolerance;
+
+    /**
+     * Relative error tolerance for adaptive time stepping.
+     * Negative values indicate to use PETSc's default.
+     */
+    double relative_tolerance;
```

Comment:
A better choice may be to enforce that the values are non-negative, and to say that zero implies PETSc's default.

Reply:
Yes, please. I much prefer being explicit than to overload the meaning of variables.